crates/wasmtime/src/runtime/component/func/typed.rs

```diff
@@ -687,6 +687,9 @@ pub unsafe trait ComponentType: Send + Sync {
     #[doc(hidden)]
     const IS_RUST_UNIT_TYPE: bool = false;
 
+    #[doc(hidden)]
+    const IS_FLAT_TYPE: bool = false;
```

Comment:
Could this be renamed to `MAY_REQUIRE_REALLOC`? This otherwise isn't a guarantee that can be relied on as it's only set for integers, but it's not set for aggregates of other flat types (e.g. option-of-enum, etc).

Also can you add documentation that getting this correct is required for not panicking? Basically how other parts of the codebase will, for example, skip fibers during lowering if "may require realloc" is `false`